### PR TITLE
Add peerDependencies

### PR DESCRIPTION
**Pull request by jcfilben**
Adjust the dependencies, move certain things to be specified as peerDependencies so we don't end up with 2 versions of react in an application

### CODE REVIEW

File: grommet-leaflet/package.json [hunk 1]
```diff
@@ -23,20 +23,26 @@
...
     "leaflet.markercluster": "^1.5.3",
-    "react": "^18.2.0",
-    "react-dom": "^18.2.0",
     "react-leaflet": "^4.2.1",
-    "styled-components": "^5.1.0",
+    "prop-types": "^15.8.1",
     "uuid": "^9.0.1"
   },
+  "peerDependencies": {
+    "grommet": "^2.33.2",
```
Comment (taysea) [inline grommet-leaflet/package.json:41]: Does it have to be this version of grommet? or are any earlier versions supported?

Comment (jcfilben) [reply grommet-leaflet/package.json:41]: Good call out, we should be supporting earlier versions as well. Just not quite sure how far to go back. I checked and the components used in grommet-leaflet are pretty basic like Box, Button, and Text. So I think we could just say `"grommet": "^2.0.0"`.

Comment (taysea) [reply grommet-leaflet/package.json:41]: Cool yeah was just curious about that. Nothing wrong with encouraging a more recent version but wasn't sure if there was anything prohibiting an earlier one. I'm good with 2.0.0 and we can adjust if necessary.

Comment (jcfilben) [reply grommet-leaflet/package.json]: Just pushed up that change

**Review by taysea — APPROVED** [review]

**Pull request merged**